### ISSUE TITLE
How can I use the MobileNet from tf.slim.net?

### ISSUE THREAD

**Issue by apollo-time** [davidsandberg/facenet#346]
I want to make face recognition model on mobile device.
I add the MobileNet model from tf.slim.net likely with squeezenet.
And I try train CASIA database with train_softmax.py.
But I got very poor result.
I got the train loss value like with squeezenet, but LFW train accuracy is very poor and very strange.
What's the matter?
How can I use the MobileNet on face recognition?

**Comment by apollo-time** [comment]
I share the tensor board, yellow is squeezenet, blue is mobilenet.
I add the only mobilenet and other is same!






![train_facenet](https://user-images.githubusercontent.com/9794932/27423497-56cb2636-5764-11e7-97ef-19a978d08355.jpg)

**Comment by davidsandberg** [comment]
Hi,
Did you get this to work? I would try to change the weight decay, maybe remove the center loss and maybe try with different learning rates.

**Comment by apollo-time** [comment]
thanks your reply.
I didn't get enough result yet.
I already try remove center loss.
How can I change the weight decay?

**Comment by dymat** [comment]
Hey @apollo-time ,

you can pass weight decay as parameter to train_softmax.py. Have a look at the end of train_softmax.py. There you find all possible parameters: https://github.com/davidsandberg/facenet/blob/master/src/train_softmax.py#L399

I also tried to use Mobilenet_v1 with facenet but as you I get poor results (but I still didn't experimented with different weight_decay values). So let me please know if you get better results in your experiments.

Best regards,
d

**Comment by apollo-time** [comment]
Hi @dymat  ,
Do you think the poor result from the incorrect weight_decay value?
I set the parameter as same with @davidsandberg train_softmax.
Is the weight_decay param sensitivity for regularization?

**Comment by dymat** [comment]
Hi @apollo-time,

I actually don't know the reason for the poor results. Instead of playing with different hyperparameters I took squeezenet and experimented a lot with residual connections in squeezenet and with weight different initialisations...

Can't help you here. Sorry.

**Comment by ugtony** [comment]
I got results similar to @apollo-time 's. I cannot train the official mobilenet model even after I manually added some residual connections to it. 
Then I try to change inception_resnet_v1 by replacing the inception modules, conv2d layers, and reduction layers with the depthwise separable layers, and the model is still trainable (got accuracy=0.961 on LFW with learning_rate=0.1, batch_size=80 after 1125*40 iterations). The input size, number of layers, and filter shape are made the same as baseline MobileNet.

**Comment by zxk2046** [comment]
@davidsandberg @ugtony  I am  trying to do face recoginition on mobile. Can we only use mobilenet classifer and fine-tune mobilenet to recoginize specific person for unlocking cellphone? Or do we need to combine mobilenet with facenet embeddings to retrain the model?

**Comment by hawkjk** [comment]
It try to use google released mobilenet and resnet to replace inception resnet v1, failed. But after using the resnetv1 modele written by myself, the result is ok. It seems that the mobilenet and resnet model released by google is difficult to config.

**Comment by JianbangZ** [comment]
@ugtony 97.3% is kinda aligned with the papaer says (1% accuracy decrease). Could you share the modified inception_resnet_v1.py with MobileNet's depthwise separable layers? How much speed are you seeing while training, is it 3-5 times faster?

**Comment by ugtony** [comment]
@JianbangZ 
Sorry, I was focusing on #385 recently, I haven't polished the code yet. 

Since I didn't trained inception-resnet-v1 on 224x224 images before, so I can't make a fair comparison between the two models. The closest resolution I've trained inception-resnet-v1 on was 200x200. The comparison below are based on that.

I didn't see much gain in training speed, the iteration is about 1.1 secs (1.7 secs before) with a K80 GPU. And the testing time is 0.089s (0.117s before) on Intel(R) Xeon(R) CPU E5-2630 0 @ 2.30GHz (with batch_size = 1). 
Some more statistics:
frozen model size: 90M(inception-resnet-v1) => 14M.
memory usage when testing process is running: 1.9G (2.5G).

As you can see, the performance gain wan't much. I'm not sure if it is reasonable.

**Comment by shanren7** [comment]
I tried training mobilenet and shufflenet, the lfw accuracy was poor(lfw: ~0.60). squeezeNet was ok(lfw: ~0.97). after replace squeezeNet's conv2d with depthwise separable convolution, network was still trainable(lfw:~0.96).  strange...

**Comment by karlTUM** [comment]
Have anyone successfully trained Mobilenet? I also tied. But my loss could not decrease.
